### PR TITLE
Fix warning about using directives in multiple files when two java files are present

### LINKED ISSUE #1786: False warning about directives in multiple java files
**Version(s)**
0.1.19

**Describe the bug**
If the compiled sources contain more than one java file, a warning about using directives in multiple files  is reported, even if there are no using directives in the source files.

**To Reproduce**
```sh
mkdir /tmp/bug-repro
cd /tmp/bug-repro
echo 'public class Foo {}' > Foo.java
echo 'public class Bar {}' > Bar.java
scala-cli compile .
```

This outputs:
```scala
[warn] ./Bar.java:1:1: Using directives detected in multiple files. It is recommended to keep them centralized in the /private/tmp/bug-repro/project.scala file.
[warn] public class Bar {}
[warn] ^
[warn] ./Foo.java:1:1: Using directives detected in multiple files. It is recommended to keep them centralized in the /private/tmp/bug-repro/project.scala file.
[warn] public class Foo {}
[warn] ^
```

**Expected behaviour**
Compilation should report no warnings

### PR DESCRIPTION
**Pull request by MaciejG604**
Resolves #1786

### CODE REVIEW

File: modules/build/src/main/scala/scala/build/preprocessing/JavaPreprocessor.scala [hunk 1]
```diff
@@ -53,7 +53,11 @@ final case class JavaPreprocessor(
               allowRestrictedFeatures
             )
           }
-          Seq(PreprocessedSource.OnDisk(
+          if (
+            updatedOptions.global == BuildOptions.monoid.zero && updatedOptions.scoped.isEmpty && updatedOptions.unused.isEmpty
```
Comment (lwronski) [inline modules/build/src/main/scala/scala/build/preprocessing/JavaPreprocessor.scala]: Move this condition to the class `DirectivesProcessorOutput`.

Comment (MaciejG604) [reply modules/build/src/main/scala/scala/build/preprocessing/JavaPreprocessor.scala]: Moved checking for no directives to the `ExtractedDirectives` class, I think it makes much more sense there.

Comment (lwronski) [reply modules/build/src/main/scala/scala/build/preprocessing/JavaPreprocessor.scala]: Good catch!

File: modules/build/src/main/scala/scala/build/preprocessing/JavaPreprocessor.scala [hunk 2]
```diff
@@ -11,10 +11,10 @@ import scala.build.Logger
 import scala.build.errors.BuildException
 import scala.build.input.{Inputs, JavaFile, SingleElement, VirtualJavaFile}
 import scala.build.internal.JavaParserProxyMaker
-import scala.build.options.BuildRequirements
+import scala.build.options.{BuildOptions, BuildRequirements}
```
Comment (lwronski) [inline modules/build/src/main/scala/scala/build/preprocessing/JavaPreprocessor.scala]: Remove unused import.

**Review by lwronski — APPROVED** [review]
LGTM

**Pull request merged**